internal/messaging/mint_v2.go

```diff
@@ -111,33 +112,36 @@ func (h *evmResponseHandler) processMintResponseV2(ctx context.Context, response
 }
 
 func (h *evmResponseHandler) getPriceAndTokenV2(ctx context.Context, price *typesv2.Price) (*big.Int, common.Address, *big.Int, error) {
+	paymentToken := booking.NativePaymentToken
 	priceBigInt := big.NewInt(0)
-	paymentToken := zeroAddress
 	isoCurrency := big.NewInt(0)
 	var err error
+
 	switch currency := price.Currency.Currency.(type) {
 	case *typesv2.Currency_NativeToken:
-		priceBigInt, err = h.bookingService.ConvertPriceToBigInt(price.Value, price.Decimals, int32(18)) // CAM uses 18 decimals
-		if err != nil {
-			return nil, zeroAddress, nil, fmt.Errorf("error minting NFT: %w", err)
-		}
+		priceBigInt, err = booking.ConvertPriceToBigInt(price.Value, price.Decimals, booking.NativeTokenDecimals)
 	case *typesv2.Currency_TokenCurrency:
+		var tokenDecimals int32
 		if !common.IsHexAddress(currency.TokenCurrency.ContractAddress) {
-			return nil, zeroAddress, nil, fmt.Errorf("invalid contract address: %s", currency.TokenCurrency.ContractAddress)
+			return nil, common.Address{}, nil, fmt.Errorf("invalid contract address: %s", currency.TokenCurrency.ContractAddress)
 		}
 		contractAddress := common.HexToAddress(currency.TokenCurrency.ContractAddress)
-		tokenDecimals, err := h.erc20.Decimals(ctx, contractAddress)
-		if err != nil {
-			return nil, zeroAddress, nil, fmt.Errorf("failed to fetch token decimals: %w", err)
-		}
-		priceBigInt, err = h.bookingService.ConvertPriceToBigInt(price.Value, price.Decimals, tokenDecimals)
+		tokenDecimals, err = h.erc20.Decimals(ctx, contractAddress)
 		if err != nil {
-			return nil, zeroAddress, nil, err
+			return nil, common.Address{}, nil, fmt.Errorf("failed to fetch token decimals: %w", err)
 		}
+
+		priceBigInt, err = booking.ConvertPriceToBigInt(price.Value, price.Decimals, tokenDecimals)
 		paymentToken = contractAddress
 	case *typesv2.Currency_IsoCurrency:
-		// For IsoCurrency, keep price as 0 and paymentToken as zeroAddress
+		priceBigInt, err = booking.ConvertPriceToBigInt(price.Value, price.Decimals, booking.ISODecimals)
+		paymentToken = booking.ISOPaymentToken
 		isoCurrency = big.NewInt(int64(currency.IsoCurrency))
 	}
+
+	if err != nil {
+		return nil, common.Address{}, nil, fmt.Errorf("failed to convert price to big.Int: %w", err)
+	}
+
 	return priceBigInt, paymentToken, isoCurrency, nil
 }
```

Comment:
Here we are setting default values for price and payment token. 
```golang
	paymentToken := booking.NativePaymentToken
	priceBigInt := big.NewInt(0)
```
This looks dangerous to me. (Also in `mint_v1.go`)

Consider the case when the message does not have a currency value set. Wouldn't this cause the switch case statement fall through silently and returning price with zero value and payment token type native? Resulting a booking token with zero price, unintentionally?

Reply:
yes, your assumption correct
lets also add default with error then